src/ambianic/pipeline/timeline.py

```diff
@@ -183,3 +189,15 @@ def get_event_log(pipeline_context: PipelineContext = None) \
         pipeline_event_log,
         {PIPELINE_CONTEXT_KEY: pipeline_context})
     return pipeline_event_log
+
```

Comment:
@bhavikapanara did you copy&paste the Encoder code from [store.py](https://github.com/ambianic/ambianic-edge/blob/6364c26e04390cc96530b475759198655fabd469/src/ambianic/pipeline/store.py#L184)? 

That's not a good coding practice. If you change the encoder in one place, it may break the code that uses the other. Please move the Encoder to an utility package and reference it as needed.

An alternative solution would be not to touch timeline.py at all. Instead slightly refactor the [following lines in store.py](https://github.com/ambianic/ambianic-edge/blob/6364c26e04390cc96530b475759198655fabd469/src/ambianic/pipeline/store.py#L107):

```
        with open(json_path, 'w', encoding='utf-8') as f:
            json.dump(save_json, f, ensure_ascii=False, indent=4,
                      cls=JsonEncoder)
        # e = PipelineEvent('Detected Objects', type='ObjectDetection')
        self.event_log.info('Detection Event', save_json)
```

Instead of json.dump directly to a file, dump to a string then:
1. Use the json string to save it in a json file and 
2.  Reuse the same string to deserialize to a python object (as you [already do that in timeline.py](https://github.com/ambianic/ambianic-edge/pull/333/files#diff-f5bb57b8d067d992fa63cf61c1728a2a032df7f7c1dfcc53e7d7de88c3799153R104)) which is passed to `self.event_log.info` (instead of the raw python object with the problematic numpy types).

This effectively shifts the responsibility of cleaning up the event object from timeline.py to store.py.

Reply:
@ivelin It's a really great idea. 
Will implement it.